From 268291317ccd1805dc1c801ff88641ba06c6a7f0 Mon Sep 17 00:00:00 2001
From: Florian Walpen <dev@submerge.ch>
Date: Sat, 20 Jan 2024 23:06:14 +0100
Subject: [PATCH] cmake: Use all pkgconf cflags for ImageMagick

Build fails with ImageMagick 7.1.1:

`# error "you should set MAGICKCORE_HDRI_ENABLE"`

The missing build flags are available from `pkgconf --cflags-only-other`
and as `${MAGICK_CFLAGS_OTHER}` in CMake. Add them to the build.
---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 64a7b32..52bce6f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -141,6 +141,12 @@ target_link_libraries (
     ${QT_LIBRARIES}
 )
 
+target_compile_options(
+    pdfmixtool
+    PUBLIC
+    ${MAGICK_CFLAGS_OTHER}
+)
+
 INSTALL(TARGETS pdfmixtool RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
 INSTALL(FILES resources/eu.scarpetta.PDFMixTool.desktop DESTINATION ${CMAKE_INSTALL_DATADIR}/applications)
 INSTALL(FILES resources/eu.scarpetta.PDFMixTool.appdata.xml DESTINATION ${CMAKE_INSTALL_DATADIR}/metainfo)
-- 
GitLab

